tensorflow_addons/layers/adaptive_pooling.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Pooling layers with fixed size outputs"""
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class AdaptiveAveragePooling2D(tf.keras.layers.Layer):
+    """Average Pooling with adaptive kernel size and strides.
+        Arguments:
+            output_shape: Tuple of integers specifying (Output Height, Output Width)
+
+        Input shape:
+            - If `data_format='channels_last'`:
+                4D tensor with shape `(batch_size, height, width, features)`.
+            - If `data_format='channels_first'`:
+                4D tensor with shape `(batch_size, features, height, width)`.
+
+        Output shape:
+            - If `data_format='channels_last'`:
+                4D tensor with shape `(batch_size, h_bins, w_bins, features)`.
+            - If `data_format='channels_first'`:
+                4D tensor with shape `(batch_size, features, h_bins, w_bins)`.
```

Comment:
Please see how core TF does doc here.

https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/keras/layers/pooling.py#L356-L367

Maybe change `pooled_rows` to `output_shape[0]` and `pooled_cols` to `output_shape[1]`.